pkg/client/client.go

```diff
@@ -268,10 +291,18 @@ func (o *OptimizelyClient) GetAllFeatureVariables(featureKey string, userContext
 		switch varType := v.Type; varType {
 		case entities.Boolean:
 			out, err = strconv.ParseBool(val)
+			errs = multierror.Append(errs, err)
 		case entities.Double:
 			out, err = strconv.ParseFloat(val, 64)
+			errs = multierror.Append(errs, err)
 		case entities.Integer:
 			out, err = strconv.Atoi(val)
+			errs = multierror.Append(errs, err)
+		case entities.JSON:
+			var intf map[string]interface{}
+			err = json.Unmarshal([]byte(val), &intf)
+			out = intf
```

Comment:
To avoid this `Unmarshal` being implemented in multiple places we can use the OptimizelyJSON object here.
```go
var json = optimizelyjson.NewOptimizelyJSONfromString(val)
out, err = json.ToMap()
```